### PR TITLE
8265606: Reduce allocations in AdapterHandlerLibrary::get_adapter

### PR DESCRIPTION
**Pull request by cl4es**
Use stack allocated arrays used when looking up and creating adapters. Resource allocate if number of args grows large, which is relatively rare.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265606](https://bugs.openjdk.java.net/browse/JDK-8265606): Reduce allocations in AdapterHandlerLibrary::get_adapter

### Reviewers
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3595/head:pull/3595` \
`$ git checkout pull/3595`

Update a local copy of the PR: \
`$ git checkout pull/3595` \
`$ git pull https://git.openjdk.java.net/jdk pull/3595/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3595`

View PR using the GUI difftool: \
`$ git pr show -t 3595`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3595.diff">https://git.openjdk.java.net/jdk/pull/3595.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/runtime/sharedRuntime.cpp [hunk 1]
```diff
@@ -2954,7 +2961,7 @@ VMRegPair *SharedRuntime::find_callee_arguments(Symbol* sig, bool has_receiver,
   // This method is returning a data structure allocating as a
   // ResourceObject, so do not put any ResourceMarks in here.
 
-  BasicType *sig_bt = NEW_RESOURCE_ARRAY(BasicType, 256);
+  BasicType sig_bt[256];
```
Comment (iklam) [inline src/hotspot/share/runtime/sharedRuntime.cpp:2964]: Was this line changed by mistake? The rest of the PR looks good to me.

**Review by iklam — CHANGES_REQUESTED** [review]

File: src/hotspot/share/runtime/sharedRuntime.cpp (hunk 1 above)
Comment (cl4es) [reply src/hotspot/share/runtime/sharedRuntime.cpp:2964]: It was intentional, but it's also a bit of a scope creep: this method is about creating and returning the VMRegPair array, but it also resource-allocs this sig_bt array and keeps it alive for however long the regs array is kept alive, which would be a memory leak if not for the fact that it's mostly used in a transient way.

Comment (iklam) [reply src/hotspot/share/runtime/sharedRuntime.cpp:2964]: The think the original code was written to avoid allocating large arrays on the stack.

Comment (cl4es) [reply src/hotspot/share/runtime/sharedRuntime.cpp]: Ok, I reverted this change. It's not part of anything I know sees much use anyhow.

**Review by iklam — APPROVED** [review]

**Pull request closed without merging**